src/components/searchMenu/searchPartComponents/DateInput.js

```diff
@@ -0,0 +1,27 @@
+import { useContext } from "react"
+import { OrderSearchContext } from "../../../providers/OrdersSearchProvider"
+import { dateToString } from "../../../utils/dateToString"
+
+export const DateInput = ({ className, name, date, labelText = "" }) => {
+    const { searchConditions, setSearchConditions } = useContext(OrderSearchContext)
+    const handleChange = (event) => {
```

Comment:
Bezpieczniej przy takich update'ach zawsze uzyc callbacka do setState i prevState'a - bedziemy mieli pewnosc ze nasz poprzedni stan na pewno jest poprzedni, jesli uzywamy bezposrednio tak jak tutaj ...searchConditions nie zawsze to jest reguła
```
const handleChange = (event) => {
        setSearchConditions(prevState => ({
            ...prevState,
            orderDate: {
                ...prevState.orderDate,
                [name]: new Date(event.target.value)
            }
        }))
    }
```